src/main/java/com/cse/ngsa/app/middleware/RequestLogger.java

```diff
@@ -34,24 +39,57 @@
       
     // process next handler
     return webFilterChain.filter(serverWebExchange).doFinally(signalType -> {
-      // compute request duration and get status code
-      long duration = System.currentTimeMillis() - startTime;
       int statusCode = serverWebExchange.getResponse().getStatusCode().value();
       
       // don't log favicon.ico 404s
       if (pathQueryString.startsWith("/favicon.ico")) {
         return;
       }
-
+      
       // don't log if log level >= warn but response code < 400
       if (logger.getLevel().isMoreSpecificThan(Level.WARN) && statusCode < 400) {
         return;
       }
+      
+      // In general, no need to check if serverWebExchabge have MS-CV attribute
+      // But a good practice to check for it anyway
+      // CorrelationVector cv;
+      // if (serverWebExchange.getAttributes().containsKey(CorrelationVector.HEADER_NAME)) {
+      //   cv = (CorrelationVector)serverWebExchange.getAttribute(CorrelationVector.HEADER_NAME);
+      // } else {
+      //   // This one should'nt happen!
+      //   LogManager.getRootLogger().error("MS-CV Attribute not found");
+      //   cv = CorrelationVectorExtensions.extend(serverWebExchange);
+      // }
```

Comment:
Create a different branch from this one that has this code or stash it in a named stash so you can remove the commented code from the PR.